src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2630,6 +2630,11 @@ def get_disable_ai_toolchain_operator(self) -> bool:
         # because it's already checked in get_ai_toolchain_operator
         return self.raw_param.get("disable_ai_toolchain_operator")
 
+    def get_ssh_access(self) -> Union[str, None]:
+        """Obtain the value of ssh_access.
+        """
+        return self.raw_param.get("ssh_access")
+
```

Comment:
From my understanding, you could omit the changes in `managed_cluster_decorator.py`, as it contains and calls the agent pool decorator by default.

Reply:
I tried revert all changes in `managed_cluster_decorator.py`, and rerun live test, but it reports error, the property is not expected.

Reply:
It failed at below position:
```
        # create new nodepool
        add_nodepool_cmd = 'aks nodepool add -g {resource_group} --cluster-name {name} -n nodepool2 ' \
                           '--ssh-access localuser ' \
                           '--aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/DisableSSHPreview'
        self.cmd(add_nodepool_cmd, checks=[
            self.check('provisioningState', 'Succeeded'),
            self.check('securityProfile.sshAccess', 'LocalUser'),
        ])
    
        # update cluster
        update_cmd = 'aks update --resource-group={resource_group} --name={name} ' \
                     '--ssh-access disabled --yes ' \
                     '--aks-custom-header AKSHTTPCustomFeatures=Microsoft.ContainerService/DisableSSHPreview'
>       self.cmd(update_cmd, checks=[
            self.check('provisioningState', 'Succeeded'),
            self.check('agentPoolProfiles[0].securityProfile.sshAccess', 'Disabled'),
            self.check('agentPoolProfiles[1].securityProfile.sshAccess', 'Disabled'),
        ])
```

After creating new nodepool and try to perform cluster level update, expected value is "Disabled", but got "LocalUser", I haven't went through the code very deeply, do you think it is acceptable if we keep current change in `managed_cluster_decorator.py`

Reply:
Ah, if you want to update all nodepools in command `az aks update`, there's a workaround like https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py#L6643-L6654 Otherwise, it would only update the default nodepool.

And I think your current method is also ok.

Reply:
Thanks! let's keep current method